core/src/main/java/org/moditect/internal/parser/JavaVersionHelper.java

```diff
@@ -0,0 +1,203 @@
+/*
+ *  Copyright 2017 - 2018 The ModiTect authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.moditect.internal.parser;
+
+import org.moditect.spi.log.Log;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Helper to extract and parse the Java version.
+ * Alternatively from Java 9 it is possible to use the API java.lang.Runtime#version().
+ *
+ * @author Fabio Massimo Ercoli
+ */
+public final class JavaVersionHelper {
+
+    public static Optional<Integer> resolveWithVersion(List<String> jdepsExtraArgs, Log log) {
```

Comment:
I think this method should just obtain the String with the candidate version value (and the log), not the list with all the jdeps invocation arguments. I.e. the extraction of that value (if present) should be done by the caller of this method.